src/python/esgcet/esgcet/config/cmip6_handler.py

```diff
@@ -65,77 +65,50 @@ def validateFile(self, fileobj):
         config = getConfig()
         projectSection = 'project:'+self.name
         min_cmor_version = config.get(projectSection, "min_cmor_version", default="0.0.0")
-
         data_specs_version = config.get(projectSection, "data_specs_version", default="master")
 
-        
-        file_cmor_version = None
```

Comment:
I think we need to keep this variable or it will go out of scope

Reply:
Could you give an exemple?
I just move `file_cmor_version=None` into the `except` statement. Because this is a bare except, this is equivalent to have `file_cmor_version=None` outside of the `try:... except:...` block.

Reply:
I think that a raw `except` usually has more problems than benefits. I'd suggest changing the code of `getAttribute` to raise one exception class (maybe `AttributeError`, or a user defined exception class) in response to all the exceptions that could arise internally under normal circumstances, and then catch this one exception in the calling code.  Anything other exception class, you allow to abort the code as an uncaught exception - for example, if you get `NotImplementedError` because of some later bug, you don't want to catch this.

Looking inside `getAttribute` (I think only the one inside the netCDF handler because the other `getAttribute` methods seem to be just a wrapper to it), it appears that the various exceptions that could arise internally would be an `AttributeError` if the attribute does not exist, an `IndexError` if the variable does not exist, and perhaps also a `TypeError` inside `cleanup_time_units` if the units attribute is not a string.  So you catch these and raise the one exception.  It's not many lines of code, and could avoid some future headache.

Reply:
I agree with you Alan, bare exception should be avoid as most as possible.
Your comment can be apply to the whole publisher and not only to this part of code.
When I open the publisher code into PyCharm, my screen turns red because it doesn't follow most of the PEP08 recommendations.

Anyway, it doesn't answer my question ; )

Reply:
I did a quick test and stand corrected regarding scoping with try/expect - it seems to be ok (perhaps earlier versions of python)  so I will merge the code as is :-)
Also agree about the bare exception.  In this case, we are dealing with a library where it isn't clear if you can check if the attribute is present or not (perhaps you can, and then we don't need try/except at all)